src/libpisp/common/pisp_utils.cpp

```diff
@@ -67,7 +67,8 @@ void compute_stride_align(pisp_image_format_config &config, int align)
 	if (PISP_IMAGE_FORMAT_compressed(config.format))
 		width = (width + 7) & ~7; // compression uses blocks of 8 samples
 
-	config.stride = compute_x_offset(config.format, width);
+	if (!config.stride)
+		config.stride = compute_x_offset(config.format, width);
```

Comment:
I wonder if this is not entirely safe, and we should perhaps do something like:

```c++
uint32_t computed_stride = compute_x_offset(config.format, width);
if (!config.stride || config.stride < computed_stride)
    config.stride = compute_x_offset(config.format, width);
````